test/edu/ucsb/nceas/metacat/doi/osti/OstiDOIServiceIT.java

```diff
@@ -49,52 +32,32 @@
 import java.util.List;
 import java.util.Map;
 
-import junit.framework.Test;
-import junit.framework.TestSuite;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
 
 /**
- * Junit tests for the OstiDOIService class.
+ * Integrating tests for the OstiDOIService class.
  * @author tao
  *
  */
-public class OstiDOIServiceIT extends D1NodeServiceTest {
-    
+public class OstiDOIServiceIT {
+
     private OstiDOIService service = null;
     private final static int MAX_ATTEMPTS = 20;
-    
-    /**
-     * Constructor
-     * @param name
-     */
-    public OstiDOIServiceIT(String name) {
-        super(name);
-    }
-    
-    /**
-     * Build the test suite
-     * 
-     * @return
-     */
-    public static Test suite() {
-        TestSuite suite = new TestSuite();
-        suite.addTest(new OstiDOIServiceIT("testPublishProcess"));
-        suite.addTest(new OstiDOIServiceIT("testPublishProcessForSID"));
-        suite.addTest(new OstiDOIServiceIT("testAutoPublishProcess"));
-        suite.addTest(new OstiDOIServiceIT("testAutoPublishProcessForSID"));
-        suite.addTest(new OstiDOIServiceIT("testUnregisteredShoulder"));
-        suite.addTest(new OstiDOIServiceIT("testPublishIdentifierPrivatePackageToPublic"));
-        suite.addTest(new OstiDOIServiceIT("testPublishIdentifierPrivatePackageToPartialPublic"));
-        return suite;
-    }
-    
+    private D1NodeServiceTest d1NodeTest;
+    private MockHttpServletRequest request;
+
     /**
      * Set up the test fixtures
      * 
      * @throws Exception
      */
     @Before
```

Comment:
None of these tests run "out of the box", unless you change the doi properties. Would you mind adding the following mocks, to fix this?

```java
    @Before
    public void setUp() throws Exception {
        Properties withProperties = new Properties();
        withProperties.setProperty("server.name", "UpdateDOITestMock.edu");
        withProperties.setProperty("guid.doi.enabled", "true");
        withProperties.setProperty("guid.doi.password", "apitest");
        withProperties.setProperty("guid.doi.username", "apitest");
        closeableMock = LeanTestUtils.initializeMockPropertyService(withProperties);
    }

    @After
    public void tearDown() throws Exception {
        closeableMock.close();
    }
```

Reply:
Hi, Matthew. This test is for the OSTI server, which is used by ESS-DIVE folks. Their password is a secret so we can't put it into the code. Even though, putting ezid one in the code is not a good idea.
This is the one of three known failure test classes. It needs to contact the OSTI server for working.

Reply:
Ok - instead, then, can we check on startup and `fail` with a good explanation message (like yours, ab), if doi enabled is `true` or the doi password has no value?

Reply:
suggested code for checks:

```java
    @Before
    public void setUp() throws Exception {
        final String passwdMsg =
            """
            \n* * * * * * * * * * * * * * * * * * *
            DOI CREDENTIALS NOT SET!
            Test requires OSTI-specific values for
            'guid.doi.username' & 'guid.doi.password'
            in your metacat-site.properties file!
            * * * * * * * * * * * * * * * * * * *
            """;
        try {
            assertFalse(passwdMsg, PropertyService.getProperty("guid.doi.password").isBlank());
            assertNotEquals("apitest", PropertyService.getProperty("guid.doi.username"));
        } catch (PropertyNotFoundException e) {
            fail(passwdMsg);
        }
        Properties withProperties = new Properties();
        withProperties.setProperty("guid.doi.enabled", "true");
        closeableMock = LeanTestUtils.initializeMockPropertyService(withProperties);
    }

    @After
    public void tearDown() throws Exception {
        closeableMock.close();
    }
```